### PR TITLE
Docs/update docstrings

### PR DESCRIPTION
**Pull request by vyasr**
## Description
Updates all docstrings to reflect new APIs in freud. Also renames the `neighbor_query` argument to system everywhere.

## Motivation and Context

Resolves: #294. Also related to #433. More importantly, provides a standard API for both passing in `(box, points)` throughout freud along with the `neighbors` argument.

## How Has This Been Tested?

Existing tests pass.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: freud/density.pyx [hunk 1]
```diff
@@ -508,12 +509,19 @@ cdef class RDF(SpatialHistogram1D):
...
             &self.thisptr.getRDF(),
             freud.util.arr_type_t.FLOAT)
 
     @Compute._computed_property
     def n_r(self):
+        """(:math:`N_{bins}`,) :class:`numpy.ndarray`: Histogram of cumulative
+        bin_counts values. More precisely, :code:`n_r[i]` is the average number
+        of points contained within a ball of radius :code:`R[i]+dr/2` centered
+        at a given :code:`query_point` averaged over all :code:`query_points`
+        in the last call to :meth:`~.compute` (or :meth:`~.accumulate`)."""
```
Comment (bdice) [inline freud/density.pyx:524]: How to write this without accumulation?
```suggestion
        computed thus far."""
```

**Review by bdice — APPROVED** [review]
One general comment and one code suggestion that I want a second opinion on:

Should array shapes always be in math font? Should 1D array shapes be written as `(3)`  or `(3,)` or ``:math:`\\left(3, \\right)` `` or ``(:math:`3`)``? If it's 2D, do we allow partial math font like ``(:math:`N_{clusters}`, 3, 3)`` ?

(We can also punt on this issue and say it's "good enough" for now.)

File: freud/density.pyx (hunk 1 above)
Comment (vyasr) [reply freud/density.pyx:524]: That seems about as good as we can do...

**Comment by vyasr** [comment]
We'll leave the commented changes for future work. For now, this is good enough.

**Pull request merged**